server/src/test/java/org/elasticsearch/action/admin/indices/get/GetIndexResponseTests.java

```diff
@@ -72,6 +72,10 @@ protected GetIndexResponse createBlankInstance() {
 
     @Override
     protected GetIndexResponse createTestInstance() {
+        return createTestInstance(false);
```

Comment:
Maybe we should use `randomBoolean()` here to make sure we continue to test typed responses? Now that we've removed `GetIndexResponse.fromXContent(XContentParser parser, boolean includeTypeName)` we can't test xContent serialization for typed responses in the usual way, so we would need to skip that or write a dedicated test.

Reply:
The problem is that the default toXContent/fromXContent methods now only support the "new" format. While we can output the legacy response format to XContent when setting `include_type_name=true` we cannot parse it. I removed that parsing method upon your request earlier, its the one we would have needed if we would add a "getWithTypes" method to the HLRC IndicesClient.
The AbstractStreamableXContentTestCase is set up in a way that toXContent/fromXContent are forming a pair, and since we don't have a second parsing method we also cannot test the roundtrip for the "old" output any longer.

Reply:
Right, we wouldn't be able to test xContent serialization, but I think it should still be possible to still test stream serialization, equals and hashCode, etc.? I was thinking maybe we could override `AbstractStreamableXContentTestCase#createXContextTestInstance` to create only a typeless instance, but allow `createTestInstance` to create both typed + untyped versions.

Reply:
I'll look into it.

Reply:
Great, that seems to work nicely. I didn't know that we could overwrite that method.